sympy/matrices/repmatrix.py

```diff
@@ -313,6 +393,69 @@ def equals(self, other, failing_expression=False):
                     rv = ans
         return rv
 
+    def clear_denoms(self, **opts):
```

Comment:
Why are the `opts` needed?

Reply:
The arguments are used to specify the domain that will be constructed which can affect the results e.g.:
```python
In [13]: e = sqrt(2) + sqrt(3) - sqrt(5 + 2 * sqrt(6))

In [14]: M = Matrix([[2, 1/(e + 1)], [1/(e + 1), 2]])

In [15]: M.clear_denoms()
Out[15]: 
⎛   ⎡                                           1              ⎤⎞
⎜   ⎢             2                ────────────────────────────⎥⎟
⎜   ⎢                                  __________              ⎥⎟
⎜   ⎢                              - ╲╱ 2⋅√6 + 5  + 1 + √2 + √3⎥⎟
⎜1, ⎢                                                          ⎥⎟
⎜   ⎢             1                                            ⎥⎟
⎜   ⎢────────────────────────────               2              ⎥⎟
⎜   ⎢    __________                                            ⎥⎟
⎝   ⎣- ╲╱ 2⋅√6 + 5  + 1 + √2 + √3                              ⎦⎠

In [16]: M.clear_denoms(extension=True)
Out[16]: 
⎛   ⎡2  1⎤⎞
⎜1, ⎢    ⎥⎟
⎝   ⎣1  2⎦⎠
```
I'm not sure what a good use of that is in this context but since a domain is being constructed it seems reasonable to expose the ways to control the construction of that domain.

Reply:
Is there any example that the results gives different result by values than the way it simplifies the result?

Reply:
I don't know. Some domains can be expensive to construct though and some domains give better simplification. Whenever domains are being used implicitly it is good to give the user some control over it (which is partly why these options exist).

Reply:
My opinion is that's that makes not very obvious to have here than applying `to_DM` and `clear_denoms` in chain

Reply:
I can remove this method. I doubt that many people would use `to_DM().clear_denoms()` instead but perhaps adding `as_numer_denom` as mentioned above would be better anyway.

Do you think it would be better just to remove this method from the PR?

I want to get the other things like `to_DM` merged but I am less interested in `Matrix.clear_denoms`.

Reply:
Yes, I think that it is more difficult to remove methods in the future due to backward compatibility. 
If we are not sure about the usage or we are only using it once, it’s better not to make public API for a while.

Reply:
Okay, I've removed it. I think that the `as_numer_denom` function I mentioned above would be a useful addition but it is a bit different from the `Matrix.clear_denoms` method that I just removed.

Reply:
I've updated the release notes